luavm/src/lib/instructions/functions.rs

```diff
@@ -11,28 +10,26 @@ pub fn closure(vm: &mut Vm, instr: u32) -> Result<(), LuaError> {
         .bytecode
         .get_function(vm.curr_func)
         .get_func_index(second_arg(instr) as usize);
-    vm.registers.last_mut().unwrap()[first_arg(instr) as usize] =
-        LuaVal::from(LuaClosure::new(index));
+    vm.registers[first_arg(instr) as usize] = LuaVal::from(LuaClosure::new(index));
     Ok(())
 }
 
 pub fn call(vm: &mut Vm, instr: u32) -> Result<(), LuaError> {
     // The closure_index method gives us an index of the bytecode.functions vector
     // where we have to "jump" in order to find the instructions of the callee.
-    let index = vm.registers.last().unwrap()[first_arg(instr) as usize].closure_index()?;
+    let index = vm.registers[first_arg(instr) as usize].closure_index()?;
     let old_func = vm.curr_func;
     vm.curr_func = index;
-    // generate a new set of registers for the new stack frame
+    // push the first `reg_num` registers to the stack, as the function will modify these
     let reg_num = vm.bytecode.get_function(vm.curr_func).reg_count();
-    let mut new_regs = Vec::with_capacity(reg_num);
-    new_regs.push(vm.registers.last().unwrap()[ENV_REG].clone());
-    for _ in 1..reg_num {
-        new_regs.push(LuaVal::new());
+    for i in 1..reg_num {
```

Comment:
Why `1..reg_num` and not `0..reg_num`?

Reply:
Because at `0` I store the `_ENV` variable, which is global and shared between all functions.